--- a/packages/nx/src/command-line/mcp/command-object.ts
+++ b/packages/nx/src/command-line/mcp/command-object.ts
@@ -4,7 +4,7 @@
 export const yargsMcpCommand: CommandModule = {
   command: 'mcp',
   describe: 'Starts the Nx MCP server.',
-  // @ts-expect-error - yargs types are outdated, refer to docs - https://github.com/yargs/yargs/blob/main/docs/api.md#commandmodule
+  // @ts-ignore - yargs types are outdated, refer to docs - https://github.com/yargs/yargs/blob/main/docs/api.md#commandmodule
   builder: async (y: Argv, helpOrVersionSet: boolean) => {
     if (helpOrVersionSet) {
       (await handleImport('./mcp', __dirname)).showHelp();
